### PR TITLE
Added workflow to release csm libs

### PR DESCRIPTION
**Pull request by harishp8889**
# Description
Added automation to release process.

# Issues
List the issues impacted by this PR:

| Issue ID |
| -------- |
|          | 

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Tested by running automation against gobrick library by forking the repo.

image

### CODE REVIEW

**Review by donatwork — DISMISSED** [review]

**Comment by shaynafinocchiaro** [comment]
Is this something we want to have to trigger manually in each repository? We could have a trigger action that will kick off the release in all the goclient libraries we have. Example of that is here: [trigger workflow](https://github.com/shaynafinocchiaro/test-common-github-actions/blob/main/.github/workflows/trigger-go-workflow.yaml), [target workflow](https://github.com/shaynafinocchiaro/test-csm-operator/blob/main/.github/workflows/go-version.yaml#L14)

File: .github/workflows/csm-release-libs.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2024 Dell Inc., or its subsidiaries. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+
+# This workflow is used to release libraries like gobrick, gofsutil etc.
+name: Release Libs
```
[the remark this answers is not in the document]
Comment (prablr79) [reply .github/workflows/csm-release-libs.yaml:10]: Yes better to call out clearly

**Comment by harishp8889** [comment]
> Is this something we want to have to trigger manually in each repository? We could have a trigger action that will kick off the release in all the goclient libraries we have. Example of that is here: [trigger workflow](https://github.com/shaynafinocchiaro/test-common-github-actions/blob/main/.github/workflows/trigger-go-workflow.yaml), [target workflow](https://github.com/shaynafinocchiaro/test-csm-operator/blob/main/.github/workflows/go-version.yaml#L14)

@shaynafinocchiaro @gallacher If we try to release everything in one shot like the above example then we can't provide the versions to be released for different repos. So we can use this approach for only major release as per our discussions.

File: .github/workflows/csm-release-libs.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      
+      - name: Create new tag
+        run: |
+          git config --global user.name 'github-actions'
+          git config --global user.email 'github-actions@github.com'
+          git tag ${{ github.event.inputs.version }}
+          git push origin ${{ github.event.inputs.version }}
+
+      - name: Create GitHub release
+        uses: actions/create-release@v1
```
[the remark this answers is not in the document]
Comment (harishp8889) [reply .github/workflows/csm-release-libs.yaml:51]: I read this repo documentation, and they have mentioned in the example here https://github.com/comnoco/create-release-action?tab=readme-ov-file#example-workflow---create-a-release about our current methodology only. I will investigate further on this repo and update it in my upcoming PR

File: .github/workflows/csm-release-libs.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version to release (major, minor, patch)'
+        required: true
+        default: 'none'
+
+jobs:
+  build-and-scan:
+    name: Build,Scan and Release
```
[the remark this answers is not in the document]
Comment (harishp8889) [reply .github/workflows/csm-release-libs.yaml]: fixed

File: .github/workflows/csm-release-libs.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (harishp8889) [reply .github/workflows/csm-release-libs.yaml]: fixed

File: .github/workflows/csm-release-libs.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+
+# This workflow is used to release libraries like gobrick, gofsutil etc.
+name: Release Libs
+
+# Invocable as a reusable workflow
+on:
+  workflow_call:
```
[the remark this answers is not in the document]
Comment (harishp8889) [reply .github/workflows/csm-release-libs.yaml:14]: updated

**Review by prablr79 — APPROVED** [review]
lgtm

**Pull request merged**